arch-snapshot/arch.md

```diff
@@ -48,9 +48,300 @@ only as detailed as to explain how the actors control the artifacts. The goal is
+central server to facilitate [Agreements](#agreement) between parties. As a result, offers must be propagated between nodes,
+and transactions are conducted through direct communication.
+
+Developers don’t need to worry about [offer propagation](#offer-propagation). The responsibility for propagating offers
+lies with the Yagna daemon, specifically its market module. The only task for the Provider Agent is to publish the offer
+on the market using its REST API. The REST endpoint returns a [subscription](#subscription) ID, which can later be used
+to listen for incoming [proposal](#proposal) events.
+
+#### 3. Monitor incoming Proposal events and negotiate an Agreement with the most promising Requestor
+
+The [Provider Agent](#provider-agent) plays a passive role in negotiations. Offers are propagated across the network and
+received by [Requestors](#requestor). The offer is matched locally on the Requestor's node with a [Demand](#demand).
+If the Requestor is interested, they respond by sending a [Proposal](#proposal) to the Provider. This triggers a Proposal
+event on the Provider's [subscription](#subscription) endpoint.
+
+Negotiation is the process of exchanging Proposals and adjusting their terms until the Requestor proposes an Agreement.
+The structure of a Proposal is identical to that of an Offer or Demand, using the same property and constraint language
+to describe the Agreement's conditions. During negotiations, certain aspects of the Agreement can be modified. While Offers
+and Demands represent the initial declaration of resources, terms, and conditions, the proposal exchange is a dynamic
+process of refining these terms to reach an optimal agreement for both parties.
+
+The negotiation stage serves several purposes:
+- Ensures that the Provider and Requestor communicate before signing an Agreement (since offer propagation doesn’t
+  require direct interaction between parties).
+- Allows both the Provider and Requestor to implement different strategies to maximize their benefits and select suitable partners.
+- Provides an opportunity for the Provider and Requestor to negotiate additional terms that weren’t included in the  
+  initial Proposals. This is possible through protocols built on top of the property language.
+
+Both the Provider and Requestor negotiate with multiple nodes simultaneously. The Requestor initiates the Agreement by
+proposing it to the Provider, who can either accept or reject the proposal. Once the Agreement is signed by both parties,
+the Requestor can begin using the resources. The Agreement remains valid until it is terminated by either party. The terms
+of termination (e.g., duration of the Agreement and conditions under which it can be terminated) are specified within the
+Agreement itself, rather than being defined by the Golem protocol.
+
+```mermaid
+---
+title: Simplified negotiations from Provider perspective
```

Comment:
This diagram is great but I'd rather save it for the technical view or remove Yagna form it. We do not need to introduce the user to individual processes in this chapter.

Reply:
Check out the new diagram.
I removed yagna, but mainly because it was making the overview more complicated. And I tried to describe it more from decisions point of view rather than API.

In general I don't agree that we shouldn't talk about yagna. And it's not equivalent to talking about processes at all, because as we already discussed there are layers and the split between agent and daemon is the exact point where the separation between those layers is. If we don't want to mention the split yet, we shouldn't talk about Provider Agent then.

And I'm not sure what word we should use instead in this case. Maybe just Provider. Provider is an actor but it was used as software component as well...

Reply:
I see that @prekucki uses word `Yagna Node` instead of yagna daemon. Maybe this would be more acceptable and would separate processes from abstract components?